test/utils/format/formatSmart.spec.ts

```diff
@@ -180,3 +180,26 @@ describe('Edge cases', () => {
     expect(formatSmart({ amount, precision: 2, decimals: 4 })).toEqual('12,345.67')
   })
 })
+
+describe('Amount is a string', () => {
+  test('12,345.67 - Precision 0 (just adjust formatting)', () => {
+    const amount = '12345.67'
+    expect(formatSmart(amount, 0)).toEqual('12,345.67')
+  })
+  test('12,345.6789 - More decimals than precision', () => {
+    const amount = '1234567.89'
+    expect(formatSmart({ amount, precision: 2, decimals: 4 })).toEqual('12,345.6789')
+  })
```

Comment:
I might be wrong, but maybe a test that has 0s left of the dot might fail? '1234500.89'

Also, nice to add some extra test with amount without decimals, or with more decimals than the precision. I feel some edge cases are still missing.

Reply:
I'll add more tests :)